### PR TITLE
Integrate the pi app with the DDS Permissions Manager

### PR DESCRIPTION
**Pull request by ocielliottc**

### CODE REVIEW

File: src/smartlock.sh [hunk 1]
```diff
@@ -51,33 +52,64 @@ while (( $# > 0 )); do
...
+    -PERM_GOV ${PERM_GOV} \
+    -PERM_PERMS ${PERM_PERMS} \
     "
 fi
 
 echo "CMD: '$CMD', SECURITY: '$SECURITY', LOCK_ID: '$LOCK', SECURITY_ARGS: '$SECURITY_ARGS'"
 
-PID_FILE=/home/pi/smartlock.pid
+function update_certs {
+  APP_PASSWORD='ZOIWcsPg6BYmLXSv'
```
Comment (plabanca) [inline src/smartlock.sh:86]: The password should be passed in

Comment (ocielliottc) [reply src/smartlock.sh:86]: This script is run by systemctl.  Maybe it should try to read it from a file?

File: src/smartlock.sh [hunk 2]
```diff
@@ -51,33 +52,64 @@ while (( $# > 0 )); do
...
+  APP_NONCE='lock'
```
[the remark this answers is not in the document]
Comment (ocielliottc) [reply src/smartlock.sh:87]: Why?  Isn't that text to differentiate the different instances of the app?

File: src/smartlock.sh (hunk 1 above)
Comment (ocielliottc) [reply src/smartlock.sh:86]: @mitza-oci Should we maybe require a password when running `smart-lock pi install` and then read the password from a file on startup?

File: src/smartlock.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (ocielliottc) [reply src/smartlock.sh:87]: According to the permissions manager, "the NONCE is an alphanumeric string used to differentiate instances of the same application."

File: src/smartlock.sh (hunk 1 above)
Comment (ocielliottc) [reply src/smartlock.sh]: I went ahead and added the password requirement for pi install.

Comment (mitza-oci) [reply src/smartlock.sh]: I think that will work.

**Pull request merged**